lib/evse_security.cpp

```diff
@@ -622,6 +625,113 @@ int EvseSecurity::get_leaf_expiry_days_count(LeafCertificateType certificate_typ
     return 0;
 }
 
+bool EvseSecurity::verify_file_signature(const std::filesystem::path& path, const std::string& signing_certificate,
+                                         const std::string signature) {
+    EVLOG_info << "Verifying file signature for " << path.string();
+
+    // calculate sha256 of file
+    FILE_ptr file_ptr(fopen(path.string().c_str(), "rb"));
+    if (!file_ptr.get()) {
+        EVLOG_error << "Could not open file at: " << path.string();
+        return false;
+    }
+    EVP_MD_CTX_ptr md_context_ptr(EVP_MD_CTX_create());
+    if (!md_context_ptr.get()) {
+        EVLOG_error << "Could not create EVP_MD_CTX";
+        return false;
+    }
+    const EVP_MD* md = EVP_get_digestbyname("SHA256");
+    if (EVP_DigestInit_ex(md_context_ptr.get(), md, nullptr) == 0) {
+        EVLOG_error << "Error during EVP_DigestInit_ex";
+        return false;
+    }
+    size_t in_length;
+    unsigned char file_buffer[BUFSIZ];
+    do {
+        in_length = fread(file_buffer, 1, BUFSIZ, file_ptr.get());
+        if (EVP_DigestUpdate(md_context_ptr.get(), file_buffer, in_length) == 0) {
+            EVLOG_error << "Error during EVP_DigestUpdate";
+            return false;
+        }
+    } while (in_length == BUFSIZ);
+    unsigned int sha256_out_length;
+    unsigned char sha256_out[EVP_MAX_MD_SIZE];
+    if (EVP_DigestFinal_ex(md_context_ptr.get(), sha256_out, &sha256_out_length) == 0) {
+        EVLOG_error << "Error during EVP_DigestFinal_ex";
+        return false;
+    }
+
+    // extract public key
+    BIO_ptr bio_signing_certificate_ptr(
```

Comment:
The only note I have here, is that I'd want to see the X509Wrapper usage, instead of the raw X509 management. It also does some checks (you can try catch when creating one), plus it has various functionality that we might want to extend for easier usage instead of raw pointer usage.

If a raw pointer is required, it can be extracted with the 'get()' function.